src/assign.c

```diff
@@ -430,7 +430,7 @@ SEXP assign(SEXP dt, SEXP rows, SEXP cols, SEXP newcolnames, SEXP values)
       if (newcolnum<0 || newcolnum>=length(newcolnames))
         error(_("Internal error in assign.c: length(newcolnames)=%d, length(names)=%d, coln=%d"), length(newcolnames), length(names), coln); // # nocov
       if (isNull(thisvalue)) {
-        warning(_("Column '%s' does not exist to remove"),CHAR(STRING_ELT(newcolnames,newcolnum)));
+        warning(_("Tried to assign NULL to column '%s', but column '%s' does not exist to remove"), CHAR(STRING_ELT(newcolnames,newcolnum)), CHAR(STRING_ELT(newcolnames,newcolnum)));
```

Comment:
Saying the column name twice feels a bit redundant, what about?

```suggestion
        warning(_("Tried to assign NULL to column '%s', but this column does not exist to remove"), CHAR(STRING_ELT(newcolnames,newcolnum)));
```

FYI, if you really need to re-use an input to a templated message, it's better to do so in the template itself: `...column '%1$s', but column '%1$s' does not...` avoids the need to call `CHAR(...)` twice, and will make it clearer to translators that both entries are the same without needing to refer back to the package source.

Reply:
I like this change, naming the column twice did feel redundant. Will need to update the tests now.